### DISCUSSION TITLE
Wechselrichter wir unerklärlicherweise abgeschaltet

### DISCUSSION THREAD

**Discussion by peff74** [hoylabs/OpenDTU-OnBattery#233] [Q&A]
Moin,

ich hatte in der vergangenen Woche 2x ein merkwürdiges Phänomen.
Beide Mal am Abend wenn keine Energie mehr produziert wird und die Batterie sich langsam entleert.
Ohne ersichtlichen Grund hat das System den Wechselrichter abgeschaltet.
Wenn ich versucht habe Ihn über die Website wieder einzuschalten wurde er sofort vom Powerlimiter wieder abgeschaltet.
22:12:25.622 > PowerMeterClass: TotalPower: 240.00
22:12:25.622 > [PowerLimiterClass::loop] dcVoltage: 43.30 Voltage Start Threshold: 46.00 Voltage Stop Threshold: 36.00 inverter->isProducing(): 1
22:12:25.622 > [PowerLimiterClass::loop] Stopping inverter...

Wenn ich den Powerlimiter dekatiere und versuche den Wechselrichter wieder einzuschalten, kommt überhaupt nichts in den Logs, auch der Wechselrichter bleibt aus.

Neustart des ESPs ohne Erfolg.
Neustart des Wechselrichters startete die Stromproduktion.

Mein System
- HM700 nur ein Channel
- Aktuelle Version von OpenDTU onBattery
- Powermeter über Json

Was mir in den Logs noch aufgefallen ist:
09:26:09.770 > [HttpPowerMeter] Getting the power of phase 1 failed. Error: Error: connection refused
09:26:09.770 > [HttpPowerMeter] Couldn't find a value with Json query "[1]"
09:26:09.770 > PowerMeterClass: TotalPower: 183.00

Erst sagt er, er bekommt keine Daten, dann hat er doch welche ?!?

**Comment by peff74** [comment]
Heute Abend habe ich wieder so ein Phenomen.

Inverter will ums verrecken nicht starten.
Habe nun den Powerlimiter angepasst, dass er im Bereich liegt das er starten müsste...
Steht auch in der Konosle, nur starten tut er nicht.
Solche Probleme hatte ich mit den anderen Versionen nicht.



21:16:31.040 > Websocket: [/vedirectlivedata][5] disconnect
21:16:31.040 > Websocket: [/huaweilivedata][5] disconnect
21:16:31.040 > Websocket: [/batterylivedata][5] disconnect
21:16:31.995 > RX Period End
21:16:31.995 > Success
21:16:32.052 > TX ActivePowerControl Channel: 40 --> 51 81 87 04 69 80 10 62 76 81 0B 00 05 DC 00 00 56 C0 7B 
21:16:34.048 > RX Period End
21:16:34.048 > All missing
21:16:34.048 > Nothing received, resend whole request
21:16:34.048 > TX ActivePowerControl Channel: 61 --> 51 81 87 04 69 80 10 62 76 81 0B 00 05 DC 00 00 56 C0 7B 
21:16:34.104 > Interrupt received
21:16:34.179 > RX Channel: 40 --> D1 81 87 04 69 81 87 04 69 81 00 00 0B 00 14 07 48 
21:16:36.147 > RX Period End
21:16:36.147 > Success
21:16:36.147 > Fetch inverter: 114181870469
21:16:36.200 > TX RealTimeRunData Channel: 75 --> 15 81 87 04 69 80 10 62 76 80 0B 00 64 63 D6 90 00 00 00 00 00 00 00 00 70 69 29 
21:16:36.314 > Interrupt received
21:16:36.417 > RX Channel: 61 --> 95 81 87 04 69 81 87 04 69 01 00 01 01 AE 00 03 00 0F 00 0B 00 02 00 00 00 03 3C 
21:16:36.468 > Interrupt received
21:16:37.000 > RX Channel: 75 --> 95 81 87 04 69 81 87 04 69 02 0D 13 00 00 30 5E 1B 67 00 00 08 FB 13 89 00 00 F2 
21:16:37.053 > Interrupt received
21:16:37.103 > RX Channel: 75 --> 95 81 87 04 69 81 87 04 69 83 00 00 00 00 00 00 00 9D 00 48 1F F2 2E 
21:16:37.165 > RX Period End
21:16:37.165 > Success
21:16:39.040 > Fetch inverter: 114181870469
21:16:39.130 > TX RealTimeRunData Channel: 3 --> 15 81 87 04 69 80 10 62 76 80 0B 00 64 63 D6 93 00 00 00 00 00 00 00 00 80 7D CE 
21:16:39.192 > Interrupt received
21:16:39.692 > RX Channel: 75 --> 95 81 87 04 69 81 87 04 69 02 0D 13 00 00 30 5E 1B 67 00 00 08 FB 13 89 00 00 F2 
21:16:39.758 > Interrupt received
21:16:39.817 > RX Channel: 3 --> 95 81 87 04 69 81 87 04 69 83 00 00 00 00 00 00 00 9D 00 48 1F F2 2E 
21:16:39.881 > RX Period End
21:16:39.881 > Middle missing
21:16:39.881 > Request retransmit: 1
21:16:39.881 > TX RequestFrame Channel: 23 --> 15 81 87 04 69 80 10 62 76 81 7B 
21:16:39.949 > Interrupt received
21:16:40.055 > RX Channel: 3 --> 95 81 87 04 69 81 87 04 69 01 00 01 01 AE 00 03 00 0F 00 0B 00 02 00 00 00 03 3C 
21:16:40.106 > RX Period End
21:16:40.106 > Success
21:16:40.176 > PowerMeterClass: TotalPower: 339.00
21:16:40.176 > [PowerLimiterClass::loop] dcVoltage: 43.00 Voltage Start Threshold: 42.90 Voltage Stop Threshold: 35.99 inverter->isProducing(): 0
21:16:40.176 > [PowerLimiterClass::loop] newPowerLimit: 75
21:16:40.176 > [PowerLimiterClass::loop] Starting up inverter...
21:16:40.176 > [PowerLimiterClass::loop] Limit Non-Persistent: 75 W
21:16:40.176 > TX PowerControl Channel: 40 --> 51 81 87 04 69 80 10 62 76 81 00 00 B0 01 8E 
21:16:40.233 > Interrupt received
21:16:40.920 > RX Channel: 23 --> D1 81 87 04 69 81 87 04 69 81 00 00 00 00 24 00 74 
21:16:42.439 > RX Period End
21:16:42.439 > Success
21:16:42.490 > TX ActivePowerControl Channel: 61 --> 51 81 87 04 69 80 10 62 76 81 0B 00 05 DC 00 00 56 C0 7B 
21:16:43.365 > Interrupt received
21:16:43.413 > RX Channel: 40 --> D1 81 87 04 69 81 87 04 69 81 00 00 0B 00 14 07 48 
21:16:44.379 > RX Period End
21:16:44.379 > Success
21:16:44.379 > Fetch inverter: 114181870469
21:16:44.438 > TX RealTimeRunData Channel: 75 --> 15 81 87 04 69 80 10 62 76 80 0B 00 64 63 D6 98 00 00 00 00 00 00 00 00 B0 0E 86 
21:16:44.844 > Interrupt received
21:16:45.385 > RX Channel: 40 --> 95 81 87 04 69 81 87 04 69 01 00 01 01 AE 00 03 00 0F 00 0B 00 02 00 00 00 03 3C 
21:16:45.442 > Interrupt received
21:16:45.506 > RX Channel: 23 --> 95 81 87 04 69 81 87 04 69 02 0D 13 00 00 30 5E 1B 67 00 00 08 F9 13 88 00 00 F1 
21:16:45.551 > Interrupt received
21:16:46.350 > RX Channel: 40 --> 95 81 87 04 69 81 87 04 69 83 00 00 00 00 00 00 00 9D 00 48 9F 72 2E 
21:16:46.402 > RX Period End
21:16:46.402 > Success
21:16:47.627 > Fetch inverter: 114181870469
21:16:47.682 > TX RealTimeRunData Channel: 3 --> 15 81 87 04 69 80 10 62 76 80 0B 00 64 63 D6 9B 00 00 00 00 00 00 00 00 40 1A 61 
21:16:47.737 > Interrupt received
21:16:47.811 > RX Channel: 23 --> 95 81 87 04 69 81 87 04 69 01 00 01 01 AE 00 03 00 0F 00 0B 00 02 00 00 00 03 3C 
21:16:47.884 > Interrupt received
21:16:47.943 > RX Channel: 40 --> 95 81 87 04 69 81 87 04 69 02 0D 13 00 00 30 5E 1B 67 00 00 08 FA 13 88 00 00 F2 
21:16:48.088 > Interrupt received
21:16:48.142 > RX Channel: 40 --> 95 81 87 04 69 81 87 04 69 83 00 00 00 00 00 00 00 9D 00 48 9E 31 6C 
21:16:48.292 > RX Period End
21:16:48.292 > Success
21:16:50.120 > PowerMeterClass: TotalPower: 339.00
21:16:50.120 > [PowerLimiterClass::loop] dcVoltage: 43.00 Voltage Start Threshold: 42.90 Voltage Stop Threshold: 35.99 inverter->isProducing(): 0
21:16:50.120 > [PowerLimiterClass::loop] newPowerLimit: 75
21:16:50.120 > [PowerLimiterClass::loop] Starting up inverter...
21:16:50.120 > [PowerLimiterClass::loop] Limit Non-Persistent: 75 W
21:16:50.120 > TX PowerControl Channel: 23 --> 51 81 87 04 69 80 10 62 76 81 00 00 B0 01 8E 
21:16:50.169 > Interrupt received
21:16:50.231 > RX Channel: 23 --> D1 81 87 04 69 81 87 04 69 81 00 00 00 00 24 00 74 
21:16:52.090 > RX Period End
21:16:52.091 > Success
21:16:52.157 > TX ActivePowerControl Channel: 40 --> 51 81 87 04 69 80 10 62 76 81 0B 00 05 DC 00 00 56 C0 7B 
21:16:52.505 > Interrupt received
21:16:52.569 > RX Channel: 23 --> D1 81 87 04 69 81 87 04 69 81 00 00 0B 00 14 07 48 
21:16:54.400 > RX Period End
21:16:54.400 > Success
21:16:54.400 > Fetch inverter: 114181870469
21:16:54.451 > TX RealTimeRunData Channel: 61 --> 15 81 87 04 69 80 10 62 76 80 0B 00 64 63 D6 A2 00 00 00 00 00 00 00 00 11 24 37 
21:16:54.565 > Interrupt received
21:16:54.624 > RX Channel: 40 --> 95 81 87 04 69 81 87 04 69 83 00 00 00 00 00 00 00 9D 00 48 9E 31 6C 
21:16:54.871 > RX Period End
21:16:54.871 > Middle missing
21:16:54.871 > Request retransmit: 1
21:16:54.871 > TX RequestFrame Channel: 75 --> 15 81 87 04 69 80 10 62 76 81 7B 
21:16:54.950 > Interrupt received
21:16:55.015 > RX Channel: 40 --> 95 81 87 04 69 81 87 04 69 01 00 01 01 AE 00 03 00 0F 00 0B 00 02 00 00 00 03 3C 
21:16:55.155 > RX Period End
21:16:55.155 > Middle missing
21:16:55.155 > Request retransmit: 2
21:16:55.155 > TX RequestFrame Channel: 3 --> 15 81 87 04 69 80 10 62 76 82 78 
21:16:55.570 > Interrupt received
21:16:55.630 > RX Channel: 40 --> 95 81 87 04 69 81 87 04 69 02 0D 13 00 00 30 5E 1B 67 00 00 08 FA 13 88 00 00 F2 
21:16:56.177 > RX Period End
21:16:56.177 > Success
21:16:57.116 > Fetch inverter: 114181870469
21:16:57.180 > TX RealTimeRunData Channel: 23 --> 15 81 87 04 69 80 10 62 76 80 0B 00 64 63 D6 A5 00 00 00 00 00 00 00 00 21 02 26

**Comment by helgeerbe** [comment]
Mal ne doofe Frage. Wie ist die Uptime (info->system)? Ich hatte mal ein ähnliche Symptom. Batterie wurde verbraucht, openDTU startet neu->Nach dem Neustart schaltet der Power Limiter den Inverter aus, da der Startwert nicht erreicht war.

**Reply by peff74** [reply]
Das ist es, wie blind kann man sein, das Ding ist neu gestartet. Danke!
Bleibt die Frage offen, warum?
Reset Reason ist Software CPU, was ja leider alles sein kann.

**Comment by MalteSchm** [comment]
Ich bin nicht sicher aber es könnten hier mehrere Probleme zusammenkommen.

Im ersten Fall ist folgendes passiert: Deine Batterie lag unterhalb des Start Thresholds. Das entladen der Batterie ist nach dem Neustart erst mal inaktiv. Deswegen startet der PL da nicht sondern möchte erst die Batterie laden.

Im zweiten Fall klingt es für mich so als ob der WR aussteigt.

Du schreibst das es nicht möglich ist ihn manuell einzuschalten.

Der Powerlimiter versucht ihn zu starten: 
```
21:16:50.120 > PowerMeterClass: TotalPower: 339.00
21:16:50.120 > [PowerLimiterClass::loop] dcVoltage: 43.00 Voltage Start Threshold: 42.90 Voltage Stop Threshold: 35.99 inverter->isProducing(): 0
21:16:50.120 > [PowerLimiterClass::loop] newPowerLimit: 75
21:16:50.120 > [PowerLimiterClass::loop] Starting up inverter...
21:16:50.120 > [PowerLimiterClass::loop] Limit Non-Persistent: 75 W
```

Wie du es beschreibst schafft auch der PL keinen start des WR.

Blinkt der WR rot? Was passiert wenn du WR Neustart im Webinterface durchführst?

**Reply by peff74** [reply]
Neustart im Webinterface hat dann dazu geführt, dass es funktioniert hat.
Also eher eine WR Problem?

**Comment by helgeerbe** [comment]
Warum die openDTU gelegentlich neu startet, kann ich nicht sagen. Leider habe ich das noch nie mitloggen können.
Unabhängig davon habe ich den WR im Verdacht, dass er Probleme hat, wenn er dauerhaft in Betrieb ist. Bei mir hat auf einmal das Setzen des Limits nicht mehr richtig funktioniert. Erst ein Neustart des WR hat hier Abhilfe geschaffen.

Als Workaround könnte ich mir ein "Zwangsneustart" um Mitternacht vorstellen. Dann wären auch die Tageserträge vom WR richtig. Allerdings müsste  damit dann auch der PL zurechtkommen.

**Reply by peff74** [reply]
Würde es was bringen, wenn ich mal die Serielle logge?
Da habe ich kein Problem mit... 

Was meinst Du mit PL?

Zudem müsste man bei vielen WR starts denke ich über eine Einschalt-Strombegrenzung nachdenken, oder?

**Reply by helgeerbe** [reply]
PL = Power Limiter

Wenn du permanent loggen kannst, wäre das super. Ich bräuchte das log vom Absturz, der OpenDTU.


Ich kenne den inneren Aufbau des Hoymiles nicht. Würde aber vermuten, dass der Neustart des Wechselrichters keine Trennung der Stromversorgung beinhaltet. Somit gehe ich davon aus, dass das hinsichtlich Einschaltstrom sicher ist.

**Reply by peff74** [reply]
Ich logge ja die Spannungen und Ströme des WR mit.
Bei einem Neustart sieht man da massive Peaks. Ob das aber nun "echte" Daten sind, die er da liefert, ist unklar.

![2023-05-20 09_03_32-Strom - Heimat - Dashboards - Grafana](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/91614910/9ed0d7b0-fb27-4c8f-b60d-4a572b644e52)


Zum Thema Abstürze, da habe ich jetzt erst einmal den WT32-eth01 gegen einen normalen ESP32 ausgetauscht um zu schauen was passiert. Mitloggen kann ich die Tage mal einrichten.

**Comment by peff74** [comment]
Hier ein weiterer Status.
Seitdem der "normale" ESP32 dran ist, lief das System deutlich stabiler 1 Crash pro Tag. Sonst gab es alle paar Stunden einen Crash.
Gestern habe ich dann noch den Power-Meter von JSON auf MQTT umgestellt. Da seitdem ich die Daten per JSON von meinem ESP am Stromzähler abrief, dieser plötzlich auch anfing nicht mehr stabil zu laufen.

Nun laufen beide über 24h stabil durch.

Was nun allerdings wieder auftritt ist, dass der WR nicht starten will. Trotz der eindringlichen Aufforderung der DTU
 
![2023-05-23 14_01_03-Window](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/91614910/a3f3dd84-b540-4d43-8c6e-88a9279e790c)

Erst ein Neustart des Inverters über die Website ließ den WR starten.
Hier mal das LOG nach dem "Neustarten" über die Website.
Vielleicht kann man da was draus erkennen.

> 14:04:07.234 > Received MQTT message on topic: zaehler/strom/energy_consumption
14:04:07.234 > PowerMeterClass: TotalPower: 208.00
14:04:08.157 > RX Period End
14:04:08.157 > Success
14:04:08.202 > Fetch inverter: 114181870469
14:04:08.251 > TX RealTimeRunData Channel: 23 --> 15 81 87 04 69 80 13 88 31 80 0B 00 64 6C AB B7 00 00 00 00 00 00 00 00 44 75 FA 
14:04:08.296 > Interrupt received
14:04:08.340 > RX Channel: 3 --> 95 81 87 04 69 81 87 04 69 01 00 01 01 AF 00 03 00 0D 00 0C 00 02 00 00 00 03 38 
14:04:08.385 > Interrupt received
14:04:08.430 > RX Channel: 3 --> 95 81 87 04 69 81 87 04 69 02 41 FD 00 00 30 5E 06 F2 00 00 09 0F 13 88 00 00 2C 
14:04:08.476 > Interrupt received
14:04:08.521 > RX Channel: 75 --> 95 81 87 04 69 81 87 04 69 83 00 00 00 00 00 00 00 A4 00 21 6F 31 CD 
14:04:08.685 > RX Period End
14:04:08.685 > Success
14:04:08.769 > TX PowerControl Channel: 40 --> 51 81 87 04 69 80 13 88 31 81 02 00 D0 00 43 
14:04:08.815 > Interrupt received
14:04:08.862 > RX Channel: 3 --> D1 81 87 04 69 81 87 04 69 81 00 00 02 00 44 01 17 
14:04:10.306 > Received MQTT message on topic: zaehler/strom/energy_consumption
14:04:10.306 > PowerMeterClass: TotalPower: 203.00
14:04:10.774 > RX Period End
14:04:10.774 > Success
14:04:12.800 > PowerMeterClass: TotalPower: 203.00
14:04:13.293 > Received MQTT message on topic: zaehler/strom/energy_consumption
14:04:13.293 > PowerMeterClass: TotalPower: 202.00
14:04:13.340 > Fetch inverter: 114181870469
14:04:13.386 > TX RealTimeRunData Channel: 61 --> 15 81 87 04 69 80 13 88 31 80 0B 00 64 6C AB BC 00 00 00 00 00 00 00 00 74 06 B2 
14:04:13.432 > Interrupt received
14:04:13.478 > RX Channel: 3 --> 95 81 87 04 69 81 87 04 69 81 0B 00 09 00 50 04 42 
14:04:13.817 > RX Period End
14:04:13.817 > Success
14:04:13.962 > [PowerLimiterClass::loop] dcVoltage: 230.40 Voltage Start Threshold: 43.00 Voltage Stop Threshold: 36.00 inverter->isProducing(): 0
14:04:13.962 > [PowerLimiterClass::loop] newPowerLimit: 75
14:04:13.962 > [PowerLimiterClass::loop] Starting up inverter...
14:04:13.962 > [PowerLimiterClass::loop] Limit Non-Persistent: 75 W
14:04:13.962 > TX PowerControl Channel: 75 --> 51 81 87 04 69 80 13 88 31 81 00 00 B0 01 20 
14:04:15.992 > RX Period End
14:04:15.992 > All missing
14:04:15.992 > Nothing received, resend whole request
14:04:15.992 > TX PowerControl Channel: 3 --> 51 81 87 04 69 80 13 88 31 81 00 00 B0 01 20 
14:04:16.038 > Interrupt received
14:04:16.083 > RX Channel: 3 --> D1 81 87 04 69 81 87 04 69 81 00 00 00 00 24 00 74 
14:04:16.244 > Received MQTT message on topic: zaehler/strom/energy_consumption
14:04:16.244 > PowerMeterClass: TotalPower: 205.00
14:04:17.994 > RX Period End
14:04:17.994 > Success
14:04:18.039 > TX ActivePowerControl Channel: 23 --> 51 81 87 04 69 80 13 88 31 81 0B 00 02 EE 00 00 ED 60 FB 
14:04:18.084 > Interrupt received
14:04:18.129 > RX Channel: 3 --> D1 81 87 04 69 81 87 04 69 81 00 00 0B 00 14 07 48 
14:04:19.316 > Received MQTT message on topic: zaehler/strom/energy_consumption
14:04:19.316 > PowerMeterClass: TotalPower: 87.00
14:04:20.194 > RX Period End
14:04:20.194 > Success
14:04:20.194 > Fetch inverter: 114181870469
14:04:20.239 > TX RealTimeRunData Channel: 40 --> 15 81 87 04 69 80 13 88 31 80 0B 00 64 6C AB C3 00 00 00 00 00 00 00 00 46 22 DB 
14:04:20.339 > Interrupt received
14:04:20.386 > RX Channel: 75 --> 95 81 87 04 69 81 87 04 69 83 00 00 00 20 03 E8 00 AA 00 01 DE 99 31 
14:04:20.720 > RX Period End
14:04:20.720 > Middle missing
14:04:20.720 > Request retransmit: 1
14:04:20.720 > TX RequestFrame Channel: 61 --> 15 81 87 04 69 80 13 88 31 81 D5 
14:04:20.772 > Interrupt received
14:04:20.817 > RX Channel: 23 --> 95 81 87 04 69 81 87 04 69 01 00 01 01 AD 00 B9 03 18 00 0C 00 02 00 00 00 03 96 
14:04:20.862 > RX Period End
14:04:20.862 > Middle missing
14:04:20.862 > Request retransmit: 2
14:04:20.862 > TX RequestFrame Channel: 75 --> 15 81 87 04 69 80 13 88 31 82 D6 
14:04:20.980 > RX Period End
14:04:20.980 > Middle missing
14:04:20.980 > Request retransmit: 2
14:04:20.980 > TX RequestFrame Channel: 3 --> 15 81 87 04 69 80 13 88 31 82 D6 
14:04:21.026 > Interrupt received
14:04:21.071 > RX Channel: 40 --> 95 81 87 04 69 81 87 04 69 02 41 FD 00 00 30 5E 00 00 00 00 09 25 13 88 02 F5 05 
14:04:21.117 > RX Period End
14:04:21.117 > Success
14:04:21.163 > TX AlarmData Channel: 23 --> 15 81 87 04 69 80 13 88 31 80 11 00 64 6C AB C3 00 00 00 00 00 00 00 00 9C 39 00 
14:04:21.889 > RX Period End
14:04:21.889 > All missing
14:04:21.889 > Nothing received, resend whole request
14:04:21.889 > TX AlarmData Channel: 40 --> 15 81 87 04 69 80 13 88 31 80 11 00 64 6C AB C3 00 00 00 00 00 00 00 00 9C 39 00 
14:04:22.287 > Received MQTT message on topic: zaehler/strom/energy_consumption
14:04:22.287 > PowerMeterClass: TotalPower: 169.00
14:04:22.665 > RX Period End
14:04:22.665 > All missing
14:04:22.665 > Nothing received, resend whole request
14:04:22.665 > TX AlarmData Channel: 61 --> 15 81 87 04 69 80 13 88 31 80 11 00 64 6C AB C3 00 00 00 00 00 00 00 00 9C 39 00 
14:04:22.710 > Interrupt received
14:04:22.755 > RX Channel: 40 --> 95 81 87 04 69 81 87 04 69 81 00 01 B0 01 00 01 00 FF 00 FF 00 00 00 00 C9 F9 95 
14:04:22.831 > PowerMeterClass: TotalPower: 169.00
14:04:23.415 > RX Period End
14:04:23.415 > Success
14:04:23.934 > [PowerLimiterClass::loop] dcVoltage: 42.90 Voltage Start Threshold: 43.00 Voltage Stop Threshold: 36.00 inverter->isProducing(): 1
14:04:23.934 > [PowerLimiterClass::loop] newPowerLimit: 75
14:04:25.015 > Fetch inverter: 114181870469
14:04:25.061 > TX RealTimeRunData Channel: 75 --> 15 81 87 04 69 80 13 88 31 80 0B 00 64 6C AB C8 00 00 00 00 00 00 00 00 76 51 93 
14:04:25.106 > Interrupt received
14:04:25.151 > RX Channel: 61 --> 95 81 87 04 69 81 87 04 69 01 00 01 01 AE 00 59 01 7F 00 0C 00 02 00 00 00 03 10 
14:04:25.199 > Interrupt received
14:04:25.246 > RX Channel: 40 --> 95 81 87 04 69 81 87 04 69 02 41 FD 00 00 30 5E 00 00 00 00 09 19 13 88 01 6E A1 
14:04:25.291 > Interrupt received
14:04:25.336 > RX Channel: 61 --> 95 81 87 04 69 81 87 04 69 83 00 00 00 10 03 E8 00 AB 00 02 64 C9 E9 
14:04:25.381 > Received MQTT message on topic: zaehler/strom/energy_consumption
14:04:25.381 > PowerMeterClass: TotalPower: 168.00
14:04:25.738 > RX Period End
14:04:25.738 > Success
14:04:25.793 > TX AlarmData Channel: 3 --> 15 81 87 04 69 80 13 88 31 80 11 00 64 6C AB C8 00 00 00 00 00 00 00 00 AC 4A 48 
14:04:25.839 > Interrupt received
14:04:25.885 > RX Channel: 61 --> 95 81 87 04 69 81 87 04 69 01 00 01 B0 01 00 01 00 FF 00 FF 00 00 00 00 20 D2 D7 
14:04:25.930 > Interrupt received
14:04:25.976 > RX Channel: 75 --> 95 81 87 04 69 81 87 04 69 82 00 02 01 07 00 00 00 00 00 00 2A 01 38 
14:04:26.517 > RX Period End
14:04:26.517 > Success
14:04:28.225 > Received MQTT message on topic: zaehler/strom/energy_consumption
14:04:28.225 > PowerMeterClass: TotalPower: 169.00
14:04:30.018 > Fetch inverter: 114181870469
14:04:30.064 > TX RealTimeRunData Channel: 23 --> 15 81 87 04 69 80 13 88 31 80 0B 00 64 6C AB CD 00 00 00 00 00 00 00 00 26 6E F9 
14:04:30.110 > Interrupt received
14:04:30.158 > RX Channel: 75 --> 95 81 87 04 69 81 87 04 69 01 00 01 01 AE 00 59 01 7F 00 0C 00 02 00 00 00 03 10 
14:04:30.203 > Interrupt received
14:04:30.248 > RX Channel: 61 --> 95 81 87 04 69 81 87 04 69 02 41 FD 00 00 30 5E 00 00 00 00 09 18 13 88 01 6D A3 
14:04:30.293 > Interrupt received
14:04:30.338 > RX Channel: 61 --> 95 81 87 04 69 81 87 04 69 83 00 00 00 10 03 E8 00 AB 00 02 20 07 63 
14:04:30.551 > RX Period End
14:04:30.551 > Success
14:04:30.599 > TX AlarmData Channel: 40 --> 15 81 87 04 69 80 13 88 31 80 11 00 64 6C AB CD 00 00 00 00 00 00 00 00 FC 75 22 
14:04:31.299 > Received MQTT message on topic: zaehler/strom/energy_consumption
14:04:31.299 > PowerMeterClass: TotalPower: 169.00
14:04:31.358 > RX Period End
14:04:31.358 > All missing
14:04:31.358 > Nothing received, resend whole request
14:04:31.358 > TX AlarmData Channel: 61 --> 15 81 87 04 69 80 13 88 31 80 11 00 64 6C AB CD 00 00 00 00 00 00 00 00 FC 75 22 
14:04:31.404 > Interrupt received
14:04:31.448 > RX Channel: 75 --> 95 81 87 04 69 81 87 04 69 01 00 01 B0 01 00 01 00 FF 00 FF 00 00 00 00 20 D2 D7 
14:04:31.494 > Interrupt received
14:04:31.542 > RX Channel: 75 --> 95 81 87 04 69 81 87 04 69 82 00 02 01 07 00 00 00 00 00 00 2A 01 38 


Was mir beim Blick in die Konsole noch aufgefallen ist:
![2023-05-23 14_02_57-Window](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/91614910/f83d2585-0afe-4de4-afbf-53d09e01884e)

Die Power-Meter-Class läuft teils recht häufig/doppelt durch, normal?

**Comment by helgeerbe** [comment]
Wie viele Abstürze hast du denn pro Tag? Sowohl bei PowerMeter über JSON als auch MQTT läuft der esp bei mir über mehrere Tage stabil.

Mich würde wirklich das Log interessieren, wenn er rebooted.

Einen zweiten Wechselrichter zum Testen hast du nicht zufällig. Das er so oft aussteigt scheint auch nicht normal zu sein. Überlege, ob ich #240 implementiere. Das würde dir eventuell helfen, solange die Macke am Wechselrichter nicht schlimmer wird.

**Reply by peff74** [reply]
Oh es 1-2  Crashes pro Tag.
Mit MQTT läuft's nun auch stabil. Ggf. habe/hatte ich da einen Bug in meinem ESP der per JSON Daten vom Stromzähler sendet.
Aber nun sende ich die Daten erst über den ESP per JSON an den IOBrocker und dort holt sich OPEN-DTU ON Battery es ab.
Weiter beobachten. Z.Z, logge ich meinen Stromzähler ESP, wenn der wieder stabil rennt, hänge ich das Logging an Deinen.

Einen anderen Wechselrichter hätte ich.
Habe auch schon überlegt den WR mal komplett Stromlos zu machen. Bewirkt das ggf. was anderes als 'Neustarten'?
